### DISCUSSION TITLE
Filter by property in a referenced document

### DISCUSSION THREAD

**Discussion by dubik** [ravendb/ravendb#17607] [Q&A] [answered]
I have following entities:
```
class Course {
    string Id;
    string Title;
    HashSet<string> Tags;
   // More properties
}

class Enrollment {
    string UserId;
    string CourseId;
    int someProgress;
}
```

List user's enrollements:
```C#
from enr in session.Query<Enrollment>()
    where enr.UserId == userId
    let c = RavenQuery.Load<Course>(enr.CourseId)
    select new {
        Id = c.Id,
        Title = c.Title,
        Tags = c.Tags,
        Progress = enr.someProgress
    }
```
This seems to work, now I want to filter results based on tags.
1. I can create an `Enrollment_Index` which will load also `Course` and add `Tags`. However in the query 
I need to access to both `Course` and `Enrollment` and I don't understand how to do that with index query.
2. Store `Course` needed info to an `Enrollment` index stored fields. Don't think this is possible,
Store() let's me choose fields from ´Enrollment` only
3. Embedd tags into ´Enrollment` (and some other properties), but this will put burden on me, to keep them
in sync. I would prefer not to do that.
4. Create and index, add `Course.Tags` to it, query only on `Enrollment`, go through results and load 
referenced `Course` data. With the help of `Include()` this probably won't need a separate db call but 
`Course` has many more fields, and it looks silly to transfer them just for a couple of fields.

Is there a better solution?

**Comment by ayende** [comment, marked as answer]
You are looking at querying related documents, this is covered here:
https://ravendb.net/docs/article-page/6.0/csharp/indexes/indexing-related-documents
